### PR TITLE
[xdai-stake-lp] Add xdai-stake-lp strategy

### PR DESCRIPTION
**Pull request by maxaleks**
This strategy returns STAKE token balances of Uniswap and Sushiswap liquidity providers

### CODE REVIEW

File: src/strategies/xdai-stake-lp/index.ts [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        internalType: 'uint256',
+        name: '',
+        type: 'uint256'
+      }
+    ],
+    payable: false,
+    stateMutability: 'view',
+    type: 'function'
+  }
+];
```
Comment (ChaituVR) [inline src/strategies/xdai-stake-lp/index.ts:46]:
```suggestion
const abi = ['function balanceOf(address owner) view returns (uint256)'];
```

File: src/strategies/xdai-stake-lp/index.ts [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+const exchanges = {
+  uniswap: {
+    lpTokenAddress: '0x3B3d4EeFDc603b232907a7f3d0Ed1Eea5C62b5f7',
+    subgraph: 'https://api.thegraph.com/subgraphs/name/uniswap/uniswap-v2'
+  },
+  sushiswap: {
+    lpTokenAddress: '0x9Fc5b87b74B9BD239879491056752EB90188106D',
+    subgraph: 'https://api.thegraph.com/subgraphs/name/sushiswap/exchange'
+  }
+};
```
Comment (ChaituVR) [inline src/strategies/xdai-stake-lp/index.ts:22]: Would be great to use these as default but can also overwrite addresses and URLs if passed from params (examples.json)

File: src/strategies/xdai-stake-lp/index.ts [hunk 3]
```diff
@@ -0,0 +1,149 @@
+import { formatUnits } from '@ethersproject/units';
+import { getAddress } from '@ethersproject/address';
+import { multicall, subgraphRequest } from '../../utils';
+import examplesFile from './examples.json';
```
Comment (ChaituVR) [inline src/strategies/xdai-stake-lp/index.ts:4]: Examples JSON imported automatically now (with new changes) :) you can remove this
```suggestion
```

File: src/strategies/xdai-stake-lp/index.ts [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+export const author = 'maxaleks';
+export const version = '0.1.0';
+export const examples = examplesFile;
```
Comment (ChaituVR) [inline src/strategies/xdai-stake-lp/index.ts:8]: Examples JSON imported automatically now (with new changes) :) you can remove this
```suggestion
```

**Review by ChaituVR — CHANGES_REQUESTED** [review]

**Comment by bonustrack** [comment]
@maxaleks Hey, actually we already have a strategies that does this, one is called "sushiswap" and "uniswap" (for V2), does your strategy add something more? If not it's preferable to use the generic strategies for it.

**Comment by maxaleks** [comment]
@bonustrack Hey, the Sushiswap strategy does not take into account that most liquidity providers send their LP tokens to the Master Chef contract. But you're right, I can update the existing Sushiswap strategy instead of creating a new one

**Comment by ChaituVR** [comment]
@maxaleks Cool then, closing the PR for now, we will wait for PR on sushiswap strategy, feel free to reopen this PR if required :) Thanks

**Comment by bonustrack** [comment]
@maxaleks That would be the best yea so we don't duplicate code. If you going to change the sushiswap strategy to include LP tokens in the master chef contract i recommend to make this optional, so it doesnt change results on proposals using currently that strategy.

**Pull request closed without merging**